content/technotes/2018-07-17-phylogram.md

```diff
@@ -0,0 +1,250 @@
+and `lapply`. 
+
+For example, to label each leaf node of the tree alphabetically we can 
+create a simple labeling function and apply it to the tree nodes recursively using
+`dendrapply`.
+
+
+```r
+set_label <- function(node){
+  if(is.leaf(node)) attr(node, "label") <- LETTERS[node]
+  return(node)
+}
+x <- dendrapply(x, set_label)
+plot(x, horiz = TRUE)
+```
+
+<img src="/img/blog-images/2018-07-17-phylogram/unnamed-chunk-6-1.png" width="1000px" style="display: block; margin: auto auto auto 0;" />
+
+
+### Applications
+
+One application motivating bi-directional conversion between phylo and
+dendrogram objects involves creating publication-quality 'tanglegrams' using
+the [dendextend](https://github.com/talgalili/dendextend) package.
+For example, to see how well the fast, alignment-free *k*-mer distance
+from the [kmer](https://CRAN.R-project.org/package=kmer) package 
+performs in comparison to the standard Kimura 1980 distance measure, 
+we can create neighbor-joining trees using each method and plot them side by side
+to check for incongruent nodes.
+
+
+
+```r
+## load woodmouse data and remove columns with ambiguities
+data(woodmouse, package = "ape")
+woodmouse <- woodmouse[, apply(woodmouse, 2, function(v) !any(v == 0xf0))]
+## compute Kimura 1980 pairwise distance matrix
+dist1 <- ape::dist.dna(woodmouse, model = "K80")
+## deconstruct alignment (not strictly necessary)
+woodmouse <- as.list(as.data.frame(unclass(t(woodmouse))))
+## compute kmer distance matrix 
+dist2 <- kmer::kdistance(woodmouse, k = 7) 
+## build and ladderize neighbor-joining trees
+phy1 <- ape::nj(dist1)
+phy2 <- ape::nj(dist2)
+phy1 <- ape::ladderize(phy1)
+phy2 <- ape::ladderize(phy2)
+## convert phylo objects to dendrograms
+dnd1 <- as.dendrogram(phy1)
+dnd2 <- as.dendrogram(phy2)
+## plot the tanglegram
+dndlist <- dendextend::dendlist(dnd1, dnd2)
+dendextend::tanglegram(dndlist, fast = TRUE, margin_inner = 5)
+```
+
+<img src="/img/blog-images/2018-07-17-phylogram/unnamed-chunk-7-1.png" width="1000px" style="display: block; margin: auto auto auto 0;" />
+
+\
+
+In this case, the trees are congruent and branch lengths are similar. 
+However, if we reduce the *k*-mer size from 7 to 6,
+the accuracy of the tree reconstruction is affected, as shown by the 
+incongruence between the original K80 tree (left) and the tree derived
+from the 6-mer distance matrix (right):
+
+
+```r
+## compute kmer distance matrix 
+dist3 <- kmer::kdistance(woodmouse, k = 6) 
+phy3 <- ape::nj(dist3)
+phy3 <- ape::ladderize(phy3)
+dnd3 <- as.dendrogram(phy3)
+dndlist <- dendextend::dendlist(dnd1, dnd3)
+dendextend::tanglegram(dndlist, fast = TRUE, margin_inner = 5)
+```
+
+<img src="/img/blog-images/2018-07-17-phylogram/unnamed-chunk-8-1.png" width="1000px" style="display: block; margin: auto auto auto 0;" />
+
+\
+
+Hopefully users will find the package useful for a range of other applications, 
+and I look forward to receiving feedback from the community. 
```

Comment:
Be specific on how you would like feedback and what kind of input is helpful. Maybe suggest people open an [issue](https://github.com/ropensci/phylogram/issues) if that's what you'd like